bitcoin-magazine/bitcoin-explained/explaining-bitcoin-addresses.md

```diff
@@ -0,0 +1,649 @@
+
+Aaron Van Wirdum: 00:20:02
+
+Okay, now back 32.
+
+Sjors Provoost: 00:20:05
+
+Yeah, so there's a set of 32 characters, and that means you're... But it's doing the same thing, right? It's again saying, okay, here's a pay to public key... Yeah, a pay to public key address. In this case, a pay to witness public key because it's using segwit but it's the same idea public key hash so it says hello and then followed by the hash of the public key
+
+Aaron Van Wirdum: 00:20:30
+
+okay so back 32 addresses what are we looking at exactly Because what we're seeing for each address, it starts with BC1 and then usually a Q and then a whole bunch of other symbols. So what does this all mean?
+
+Sjors Provoost: 00:20:43
+
+That's right, so there is something called the human readable part. And that doesn't really have any meaning other than that humans can recognize, okay, if the address starts with BC, then it refers to Bitcoin. And the software of course can see this too, but both humans and software can understand this.
+
+Aaron Van Wirdum: 00:20:58
+
+Yeah, so if Litecoin would want to use these kinds of addresses. Maybe they do actually, I don't know.
+
+Sjors Provoost: 00:21:04
+
+Probably, then they might start with LT.
+
+Aaron Van Wirdum: 00:21:06
+
+Exactly. So these first two letters just refer to which currency is this about? What blockchain is this for?
+
+Sjors Provoost: 00:21:12
+
+Yeah, and it can be, I think, a fairly arbitrary number of letters. The idea is that it's separated by a one.
+
+Aaron Van Wirdum: 00:21:17
+
+Oh, it could be more than two letters as well?
+
+Sjors Provoost: 00:21:19
+
+Yeah, I think initially Bitcoin Cash was using a much longer introduction.
+
+Aaron Van Wirdum: 00:21:23
+
+I see, okay.
+
+Sjors Provoost: 00:21:24
+
+So that's pretty arbitrary. Obviously, you want to conserve space, so BC is nice and short and a one, that's a separator, has no value. So if you look at the, what do all the 32 numbers mean, then one is not in it.
+
+Aaron Van Wirdum: 00:21:36
+
+One just means... Skip this. The human readable part is over, now the fun stuff starts. Now the meat and potatoes.
+
+Sjors Provoost: 00:21:43
+
+ Right. And the fun stuff, it's a little bit easier actually than with Base58, because there's a convention that says if it's... Well, the convention is it starts with the SegWit version, so the first version of SegWit is 0, which in BEC32 is written as Q. And then it's either followed by 20 bytes or 32 bytes. And that is, then it means either it's the public key hash, or it is the script hash. And they're different lengths now because SegWit uses the SHA-256 hash of the script, rather than the RIPMAT-160 hash of the script. So in base58 the script hash is the same length as the public key hash but in segwit they're not the same length. So Simply by looking at how long the address is, you know whether you're paying to a script or you're paying to a public key. Yeah, sure, we don't have to say it.
```

Comment:
```suggestion
 Right. And the fun stuff, it's a little bit easier actually than with Base58, because there's a convention that says if it's... Well, the convention is it starts with the SegWit version, so the first version of SegWit is 0, which in BECH32 is written as Q. And then it's either followed by 20 bytes or 32 bytes. And that is, then it means either it's the public key hash, or it is the script hash. And they're different lengths now because SegWit uses the SHA-256 hash of the script, rather than the RIPEMD-160 hash of the script. So in base58 the script hash is the same length as the public key hash but in segwit they're not the same length. So Simply by looking at how long the address is, you know whether you're paying to a script or you're paying to a public key. Yeah, sure, we don't have to say it.
```